healthcareai/develop_supervised_model.py

```diff
@@ -412,31 +384,23 @@ def knn(self, scoring_metric='roc_auc', hyperparameter_grid=None, randomized_sea
 
         return algorithm.fit(self.X_train, self.y_train)
 
-    def SGDClassifier(self, scoring_metric='roc_auc', hyperparameter_grid=None, randomized_search=True):
-        # TODO
-        # KNN, gradient boosted trees, bootstrap aggregation
-        # impact coding
-        #
-        # enumerate, document and validate scoring options
-        # provide sensible defaults in neighbor list http://scikit-learn.org/stable/modules/model_evaluation.html#common-cases-predefined-values
-
-        """
-        A light wrapper for Sklearn's KNN that performs randomized search over a default (and overideable)
-        hyperparameter grid.
-        """
+    def stochastic_gradient_descent_classifier(self, scoring_metric='roc_auc', hyperparameter_grid=None,
+                                               randomized_search=True):
         if hyperparameter_grid is None:
-            loss_list = ['hinge','log']
-            penalty_list = ['l1','l2']
-            alpha_list = [0.0001,0.001,0.01,0.1]
-            hyperparameter_grid = {'loss':loss_list,'penalty':penalty_list,'alpha':alpha_list}
-            
+            # TODO add sensible SGD classifier hyperparameter grid
+            loss_list = ['hinge', 'log']
+            penalty_list = ['l1', 'l2']
+            alpha_list = [0.0001, 0.001, 0.01, 0.1]
+            hyperparameter_grid = {'loss': loss_list, 'penalty': penalty_list, 'alpha': alpha_list}
+
```

Comment:
Let's talk about SGD. `hinge` gives SVM; `log` gives logistic regression. Not sure we want to do both within one hyperparameter search. The hyperparameter search here should focus on one representation to make things straight-forward for users. See other comment.